libraries/botframework-config/src/botConfiguration.ts

```diff
@@ -0,0 +1,230 @@
+    };
+
+    public name: string = '';
+    public description: string = '';
+    public services: IConnectedService[] = [];
+    public secretKey = '';
+
+    constructor() {
+    }
+
+    public static fromJSON(source: Partial<IBotConfiguration> = {}): BotConfiguration {
+        let { name = '', description = '', secretKey = '', services = [] } = source;
+        services = services.slice().map(BotConfiguration.serviceFromJSON);
+        const botConfig = new BotConfiguration();
+        Object.assign(botConfig, { services, description, name, secretKey });
+        return botConfig;
+    }
+
+    public toJSON(): Partial<IBotConfiguration> {
+        const { name, description, services, secretKey } = this;
+        return { name, description, services, secretKey };
+    }
+
+    public static async loadBotFromFolder(folder?: string, secret?: string): Promise<BotConfiguration> {
+        let files = Enumerable.fromSource(await fsx.readdir(folder || process.cwd()))
+            .where(file => path.extname(<string>file) == '.bot');
+
+        if (files.any()) {
+            return await BotConfiguration.load(<string>files.first(), secret);
+        }
+        throw new Error(`Error: no bot file found in ${folder}. Choose a different location or use msbot init to create a .bot file."`);
+    }
+
+    // load the config file
+    public static async load(botpath: string, secret?: string): Promise<BotConfiguration> {
+        let bot = BotConfiguration.fromJSON(JSON.parse(await txtfile.read(botpath)));
+        bot.internal.location = botpath;
+
+        let hasSecret = (bot.secretKey && bot.secretKey.length > 0);
+        if (hasSecret)
+            bot.decrypt(secret);
+
+        return bot;
+    }
+
+    // save the config file
+    public async save(botpath?: string, secret?: string): Promise<void> {
+        if (!!secret) {
+            this.validateSecretKey(secret);
+        }
+
+        let hasSecret = (this.secretKey && this.secretKey.length > 0);
+
+        // make sure that all dispatch serviceIds still match services that are in the bot
+        for (let service of this.services) {
+            if (service.type == ServiceTypes.Dispatch) {
+                let dispatchService = <IDispatchService>service;
+                dispatchService.serviceIds = Enumerable.fromSource(dispatchService.serviceIds)
+                    .where(serviceId => Enumerable.fromSource(this.services).any(s => s.id == serviceId))
+                    .toArray();
+            }
+        }
+
+        if (hasSecret)
+            this.encrypt(secret);
+
+        await fsx.writeJson(botpath || <string>this.internal.location, this.toJSON(), { spaces: 4 });
+
+        if (hasSecret)
+            this.decrypt(secret);
+    }
+
+    public clearSecret() {
+        this.secretKey = '';
+    }
+
+    // connect to a service
+    public connectService(newService: IConnectedService): void {
+        if (Enumerable.fromSource(this.services)
```

Comment:
This can be simplified using the following: 
```js
const alreadyConnected = this.services.some(service => service.type === newService.type && service.id === newService.id
```
with the added benefit of not needing a dependency.